cf-file-scanner/deploy.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+#
+# Script to prepare and deploy function.
+
+set -e
+
+if (( $# != 1 )); then
+  echo 'usage: deploy.sh <GCP_PROJECT>'
+  exit 1
+fi
+
+PROJECT_ID="$1"
+SECRET_ID='cf-file-scanner'
+TOPIC_NAME='cf-file-scanner'
+CF_NAME='cf-file-scanner'
+
+./build-clamav.sh
+
+echo 'Rendering config file...'
+gcloud --project=$PROJECT_ID secrets versions access latest \
+  --secret=$SECRET_ID > application.conf
+
+echo 'Uploading function...'
+echo 'Note: it might take a few minutes to upload clamav binaries.'
+
+echo ''
+gcloud --project=$PROJECT_ID functions deploy $CF_NAME \
+  --entry-point org.broadinstitute.ddp.cf.FileScanner \
```

Comment:
Please set up a dedicated Service Account for this function - it's not great to run it with the "default" SA, which has Editor in the project (and thus defeats the purpose of an isolated function, because if the file is "malicious enough" it could possibly break out of ClamAV jail and wreak havoc in your project..)

Reply:
This SA should have only Storage Object Viewer role on the bucket, and PubSub Publisher role on the target topic.

Reply:
Yup, that's my next step!

Reply:
Done! I added a new SA with no permissions or roles. Then added the necessary storage and pubsub permissions at the resource-level.